python/pynessie/model.py

```diff
@@ -123,6 +124,38 @@ class ContentKey:
 
     elements: List[str] = desert.ib(fields.List(fields.Str))
 
+    def to_string(self: "ContentKey") -> str:
+        """Convert this key to friendly CLI string."""
+        # false positives in pylint
+        # pylint: disable=E1133
+        return ".".join(f'"{i}"' if "." in i else i for i in self.elements)
+
+    def to_path_string(self: "ContentKey") -> str:
+        """Convert this key to a url encoded path string."""
+        # false positives in pylint
+        # pylint: disable=E1133
+        return ".".join(i.replace(".", "\00") if i else "" for i in self.elements)
+
+    @staticmethod
+    def from_path_string(key: str) -> "ContentKey":
+        """Convert from path encoded string to normal string."""
+        return ContentKey([i for i in ContentKey._split_key_based_on_regex(key) if i])
```

Comment:
In that Python, it will just create an `ContentKey` with empty elements which I think shouldn't happen as the Java API requires non-empty elements, I think the Python should behave similar to its Java counterpart. 
And true, this should be addressed in both Java API and Python API to prevent empty `key` from creating empty elements.
I have created this issue to address this in both APIs https://github.com/projectnessie/nessie/issues/2648